### ISSUE TITLE
Suggestion: Circular Reference Genomes

### ISSUE THREAD

**Issue by MatFish** [CFIA-NCFAD/nf-virontus#5]
Not sure how difficult it would be to add, but it might be nice to add a feature for circular reference genomes so it will map a single long-read across the linearization point. Not sure how it would handle it now, perhaps just map a portion of the read?

**Comment by MatFish** [comment]
I ran it with my linearized circular genome and it did seem to map reads on both ends of the genome. Not sure if reads that crossed the linearization point were just mapped to one end or the other. Here is an image of the coverage plot generated by the pipeline for one of my samples.

![image](https://user-images.githubusercontent.com/22106855/101520122-a5ae6e00-3949-11eb-8e12-e58723d38825.png)

**Comment by peterk87** [comment]
Hi @MatFish that's a lot of red! Are you expecting the red section to be absent?

I think a rotate mode could be implemented so that the reads could be mapped across the ends. This is something Unicycler does. Unfortunately, I don't think Minimap2 has a circular reference sequence mode. 

For this particular sample you could try slicing at position 1000 or 1500 of the reference and concatenating the ends together. Or concatenating a couple copies of your reference sequence together. Hopefully that'd work.

**Comment by MatFish** [comment]
Yeah, this is just an amplicon I was using to check a portion of a novel genome so the coverage region is what I expected. 

The region I am concerned with it between 2000-2500 so I think this actually worked well for me. The idea just came into my head as a possible suggestion that could be helpful for future work with circular genomes. I think there are a lot of work arounds that we could do for it, but if it's something that isn't hard to implement it might be handy.
